meroxa/cli
# Run & Deploy Changes for the Turbine-core Implementation on Turbine-PY

## Description of change

CLI changes for turbine-core implementation on turbine-py, works along with changes made to https://github.com/meroxa/turbine-py/pull/219

Fixes: 
https://github.com/meroxa/cli/issues/694
https://github.com/meroxa/cli/issues/693

## Type of change




## How was this tested?

To test the following code with run command - 

1.  Grab latest changes to turbine-py and modify setup.config to point to pkg/ code (turbine-core implementation) instead of src/ code (old code) 
2. Run - `pip install -U turbine-py/ ` on your turbine-py
3. You should be able to now execute new core commands, such as `run` 
4. Install these cli changes with 'make install' on this branch 
5. Create a data app and run "meroxa app run ." 

To test the following code with deploy command - 

If this `[PR](https://github.com/meroxa/turbine-py/pull/219)` isnt merged at the time of testing and  this [PR](https://github.com/meroxa/turbine-core/pull/128) isnt merged at the time of testing, grab them both and do the following: 

We need to basically force the data app to use this version of turbine-py, instead of pip grabbing latests published version 
1. In your turbine-py repo, find requirements.txt and comment out any turbine-py requirements 
2. Modify Dockerfile to` pip install install -U turbine-py/`
3.  Run - `pip install -U turbine-py/ ` on your turbine-py 
4.  Run 'meroxa app init app --lang py' to create a new app 
5. The app generated should match the template from https://github.com/meroxa/turbine-core/pull/128 
6. Then grab the entire folder of turbine-py repo and drop it into your data app 
7. Then run 'meroxa app deploy .' 


- [ ]  Tested in minikube

## Demo

Slack - https://meroxa.slack.com/archives/C03F15EKFU3/p1682636513287819
